pyNN/brian/standardmodels/electrodes.py

```diff
@@ -95,24 +95,30 @@ def inject_into(self, cell_list):
             if not cell.celltype.injectable:
                 raise TypeError("Can't inject current into a spike source.")
         self.cell_list.extend(cell_list)
+        self.prev_amp_dict = {}
         for cell in cell_list:
-            self.indices.extend([cell.parent.id_to_index(cell)])
+            cell_idx = cell.parent.id_to_index(cell)
+            self.indices.extend([cell_idx])
+            self.prev_amp_dict[cell_idx] = 0.0
 
     def _update_current(self):
         # check if current timestamp is within dt/2 of target time; Brian uses seconds as unit of time
         if self.running and abs(simulator.state.t - self.times[self.i] * 1e3) < (simulator.state.dt/2.0):
             for cell, idx in zip(self.cell_list, self.indices):
                 if not self._is_playable:
-                    cell.parent.brian_group.i_inj[idx] = self.amplitudes[self.i] * ampere
+                    cell.parent.brian_group.i_inj[idx] += self.amplitudes[self.i] - self.prev_amp_dict[idx] #* ampere
```

Comment:
Why remove the multiplication by `ampere`? Doesn't Brian need consistent units?

Reply:
Apparently the multiplication by `ampere` is not necessary, as everything works fine. And with the updated implementation of setting currents, adding this becomes an issue as `cell.parent.brian_group.i_inj[idx]` is a float and thus cannot add the initial `ampere` value to this.

I now re-tested on the existing PyNN master code, and find that it made no difference by removing the multiplication with `ampere`.

Reply:
`float(ampere)` evaluates to 1.0, so it is not essential. However, I think it is helpful to anyone reading the code to retain this multiplication, and being explicit about units helps to avoid errors. On the other hand, since Brian 1 is little used, I'm ok with skipping it here. I would be in favour of retaining it in the equivalent Brian 2 code, however.

Reply:
Sure. Infact I have already incorporated this into Brian2 (brian2_dev branch) with the proper units.